Waidhoferj/CSC-308-309-Project
# Braden50/leave groups

Solves #125 

Added the ability to leave groups. 
Added a mutation to join groups.

To test:

- Boot up the local backend and frontend
- Go to a group page for a user that is a member. 
(Once I rebase with john's PR, we can test this better on the front end, but for now, the user state is fixed and there is only one group to leave)
- Attempt to leave the group with the "leave group" button.
- Ensure that you were brought to the group list page with an updated groups list.

To better test (optional):
Check out the new tests I added to testing.py and try to recreate them in the graphql interface. This is quite tedious, so let me know if you need help. This is not too necessary given that I already test for it; however, group membership is saved in two places, so an inconsistency would be possible and hard to catch. Therefore, it would be useful to get a second check from you guys to make sure we don't run into issues down the road.
